src/custom-elements.js

```diff
@@ -28,18 +28,21 @@ if (!priorCustomElements || priorCustomElements['forcePolyfill']) {
   PatchNode(internals);
   PatchElement(internals);
 
+  // The main document is always associated with the registry.
+  document.__CE_hasRegistry = true;
+
   /** @type {!CustomElementRegistry} */
   const customElements = new CustomElementRegistry(internals);
 
   // If `customElements.documentReady` exists and is a Promise, prevent calls to
   // define from causing upgrades until `documentReady` has resolved.
-  if (priorCustomElements) {
+  if (priorCustomElements && priorCustomElements['documentReady']) {
     const documentReady = priorCustomElements['documentReady'];
-    if (documentReady && documentReady.then instanceof Function) {
+    if (documentReady.then instanceof Function) {
```

Comment:
FYI, DOM operations that would cause an element to be constructed or a tree to be crawled for upgrades (`importNode`, `createElement`, setting `innerHTML`) that occur when this delayed-document-walk thing is happening will still happen. The definition itself hasn't been blocked, just the associated full-document walk and `whenDefined` promise resolution.